surveys/micro/2025-06-compiler-performance.md

```diff
@@ -0,0 +1,224 @@
+
+## User classification
+
+### Do you use Rust?
+
+Type: select one
+
+- Yes, I use Rust [`NEXT`](#do-you-code-in-rust-at-a-company)
+- No, I don't currently use Rust, but I have in the past [`NEXT`](#were-long-compilation-times-the-primary-reason-why-you-stopped-using-rust)
+- No, I have never used Rust [`END`](#other)
+
+### Were long compilation times the primary reason why you stopped using Rust?
+
+Type: select one
+
+- Yes
+- No, but it was one of the reasons why I stopped using Rust
+- No
+
+### Which operating systems do you use regularly for Rust development?
+
+**Note**: this is specifically about which systems you personally use for development, *not* all the
+systems you target.
+
+Type: select all that apply (optional)
+
+- Linux
+- Windows
+- Windows Subsystem for Linux
+- macOS
+- Other (open response)
+
+### Do you code in Rust at a company?
+
+If yes, what is the approximate size of the company?
+
+Type: select one (optional)
+
+- No, I do not code in Rust in a company
+- Yes (1-50 employees)
+- Yes (51-500 employees)
+- Yes (501-5000 employees)
+- Yes (more than 5000 employees)
+
+## Workflows
+
+### How do you build your Rust projects?
+
+Type: select all that apply (optional)
+
+- I use Cargo
+- I use some other build system
+- I combine Cargo and another build system
+- If you use Cargo with (or just use) other build systems, which ones do you use? (open response)
+
+### Which compilation workflows are the most important to you?
+
+Please rate how important are the following compilation workflows to you, and how much do you currently struggle with their performance currently.
+
+Type: matrix (optional)
+
+Workflows:
+
+- Unoptimized rebuild (change code, rebuild without optimizations)
+- Optimized rebuild (change code, rebuild with optimizations)
+- Workspace rebuild (change a crate which causes multiple other crates in your workspace to be rebuilt)
+- Clean unoptimized build (build a crate graph from scratch)
+- Clean optimized build (build a crate graph from scratch)
+- Build in a CI workflow
+
+Priority:
+
+- Not important for me
+- Important for me, compile times are okay for me
+- Important for me, compile times are a blocker for me
+
+### How do you primarily examine compilation errors in your code?
+
+Type: select all that apply (optional)
+
+- Using code editor (e.g. inline annotations)
+- Using terminal commands (e.g. `cargo check`)
+- Other (open response)
+
+### If you use Cargo, how often do you use the following commands after making a change to Rust code?
+
+Type: matrix (optional)
+
+Commands:
+
+- `cargo check`
+- `cargo build` or `cargo run`
+- `cargo test`
+
+Frequency:
+
+- Rarely
+- Sometimes
+- Often
+
+### How long do you need to wait for the compiler to rebuild your code after making a change?
+
+Please select the longest duration range amongst the projects that you regularly work on.
```

Comment:
Does "change" here imply this is just about local development or CI too? I'd give quite different answers for these.

Reply:
I was thinking local changes only. I will clarify, thanks.